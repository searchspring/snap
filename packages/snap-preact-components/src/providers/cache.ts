export { CacheProvider } from '@emotion/react';
import createCache from '@emotion/cache';

export const cache = createCache({
	key: 'ss',
	prepend: true,
});
