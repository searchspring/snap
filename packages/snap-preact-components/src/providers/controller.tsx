import { jsx } from '@emotion/react';
import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractController } from '@searchspring/snap-controller';

const ControllerContext = createContext<null | AbstractController>(null);

export const ControllerProvider = ({ children, controller }: { children: ComponentChildren; controller: AbstractController }) => {
	return <ControllerContext.Provider value={controller}>{children}</ControllerContext.Provider>;
};

export const useController = () => useContext(ControllerContext);

export const withController = (Component: ComponentType) => (props: any) => {
	return <Component controller={useController()} {...props} />;
};
