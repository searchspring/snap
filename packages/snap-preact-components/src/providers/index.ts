export * from './store';
export * from './theme';
