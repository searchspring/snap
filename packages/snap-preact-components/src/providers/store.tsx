import preact, { h, createContext, ComponentChildren, ComponentType } from 'preact';
import { useContext } from 'preact/hooks';
import type { AbstractStore } from '@searchspring/snap-store-mobx';

const StoreContext = createContext<null | AbstractStore>(null);

export const StoreProvider = ({ children, store }: { children: ComponentChildren; store: AbstractStore }) => {
	return <StoreContext.Provider value={store}>{children}</StoreContext.Provider>;
};

export const useStore = () => useContext(StoreContext);

export function withStore<C extends ComponentType>(Component: C): C {
	return ((props: any) => <StoreContext.Consumer>{(store) => <Component {...props} store={store} />}</StoreContext.Consumer>) as C;
}
