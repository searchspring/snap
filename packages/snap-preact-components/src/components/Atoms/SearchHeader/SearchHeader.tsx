/** @jsx jsx */
import { Fragment, h } from 'preact';

import { jsx, css } from '@emotion/react';
import { observer } from 'mobx-react-lite';

import { Theme, useTheme, CacheProvider } from '../../../providers';
import { ComponentProps, StylingCSS } from '../../../types';
import type { SearchController } from '@searchspring/snap-controller';
import { mergeProps } from '../../../utilities';
import { SearchMerchandisingStore, SearchPaginationStore, SearchQueryStore } from '@searchspring/snap-store-mobx';
import classnames from 'classnames';
const CSS = {
	searchheader: ({}: Partial<SearchHeaderProps>) => css({}),
};

export const SearchHeader = observer((properties: SearchHeaderProps): JSX.Element => {
	const globalTheme: Theme = useTheme();

	const landingPage = properties.controller?.store.merchandising.landingPage || properties.merchandisingStore?.landingPage;

	const pagination = properties.controller?.store.pagination || properties.paginationStore;
	const search = properties.controller?.store.search || properties.queryStore;

	const defaultProps: Partial<SearchHeaderProps> = {
		titleText: `Showing ${
			pagination?.multiplePages ? `<span class="ss__search-header__results-count-range"> ${pagination?.begin} - ${pagination?.end} of </span>` : ''
		} 
		<span class="ss__search-header__results-count-total">${pagination?.totalResults}</span> 
		result${pagination?.totalResults == 1 ? '' : 's'} 
		${search?.query ? `for <span class="ss__search-header__results-query">"${search.query.string}"</span>` : ''}
	`,
		correctedQueryText: `No results found for <em>"${search?.originalQuery?.string}"</em>, showing results for <em>"${search?.query?.string}"</em> instead.`,
		didYouMeanText: `Did you mean <a href=${search?.didYouMean?.url.href}>${search?.didYouMean?.string}</a>?`,
		noResultsText: `${
			search?.query
				? `<span>
			No results for <span class="ss__search-header__results-query">"${search.query.string}"</span> found.
		</span>`
				: `<span>No results found.</span>`
		}`,
	};

	const props = mergeProps('searchHeader', globalTheme, defaultProps, properties);
	const { disableStyles, style, styleScript, className } = props;

	let { titleText, subtitleText: subTitleText, correctedQueryText, noResultsText, didYouMeanText } = props;

	const styling: { css?: StylingCSS } = {};
	const stylingProps = props;

	if (styleScript && !disableStyles) {
		styling.css = [styleScript(stylingProps), style];
	} else if (!disableStyles) {
		styling.css = [CSS.searchheader(stylingProps), style];
	} else if (style) {
		styling.css = [style];
	}

	const data: data = {
		pagination: pagination,
		search: search,
	};

	if (typeof titleText == 'function') {
		titleText = titleText(data);
	}
	if (typeof subTitleText == 'function') {
		subTitleText = subTitleText(data);
	}
	if (typeof correctedQueryText == 'function') {
		correctedQueryText = correctedQueryText(data);
	}
	if (typeof noResultsText == 'function') {
		noResultsText = noResultsText(data);
	}
	if (typeof didYouMeanText == 'function') {
		didYouMeanText = didYouMeanText(data);
	}

	return (
		<CacheProvider>
			<header {...styling} className={classnames('ss__search-header', className)}>
				{landingPage ? (
					<h3 className={classnames('ss__search-header__title', 'ss__search-header__title--landing-page')}>{landingPage.title}</h3>
				) : (
					<Fragment>
						{pagination?.totalResults ? (
							<>
								<h3
									className={classnames('ss__search-header__title', 'ss__search-header__title--results')}
									aria-atomic="true"
									aria-live="polite"
									aria-label={`Now showing ${pagination.totalResults} items in the product grid`}
									dangerouslySetInnerHTML={{ __html: titleText as string }}
								></h3>

								{search?.originalQuery && (
									<h5
										className={classnames('ss__search-header__title', 'ss__search-header__title--corrected')}
										aria-atomic="true"
										aria-live="polite"
										aria-label={`No results found for ${search.originalQuery?.string}, showing results for ${search.query?.string} instead`}
										dangerouslySetInnerHTML={{ __html: correctedQueryText as string }}
									></h5>
								)}
							</>
						) : (
							pagination?.totalResults === 0 && (
								<div className="ss__search-header__no-results-wrapper">
									<h3
										className={classnames('ss__search-header__title', 'ss__search-header__title--no-results')}
										aria-atomic="true"
										aria-live="polite"
										aria-label={`No results found for ${search?.query?.string}`}
										dangerouslySetInnerHTML={{ __html: noResultsText as string }}
									></h3>

									{search?.didYouMean && (
										<h4
											className={classnames('ss__search-header__title', 'ss__search-header__title--dym')}
											aria-atomic="true"
											aria-live="polite"
											dangerouslySetInnerHTML={{ __html: didYouMeanText as string }}
										></h4>
									)}
								</div>
							)
						)}

						{subTitleText && (
							<h4
								className={classnames('ss__search-header__title', 'ss__search-header__title--subtitle')}
								aria-atomic="true"
								aria-live="polite"
								dangerouslySetInnerHTML={{ __html: subTitleText as string }}
							></h4>
						)}
					</Fragment>
				)}
			</header>
		</CacheProvider>
	);
});

export interface SearchHeaderProps extends ComponentProps {
	controller?: SearchController;
	queryStore?: SearchQueryStore;
	paginationStore?: SearchPaginationStore;
	merchandisingStore?: SearchMerchandisingStore;

	titleText?: string | ((data: data) => string);
	subtitleText?: string | ((data: data) => string);
	correctedQueryText?: string | ((data: data) => string);
	noResultsText?: string | ((data: data) => string);
	didYouMeanText?: string | ((data: data) => string);
}

interface data {
	pagination?: SearchPaginationStore;
	search?: SearchQueryStore;
}
