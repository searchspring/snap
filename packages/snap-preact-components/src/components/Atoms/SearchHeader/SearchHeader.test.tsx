import { h } from 'preact';
import { render, waitFor } from '@testing-library/preact';
import { SearchHeader } from './SearchHeader';
import { Theme, ThemeProvider } from '../../../providers';
import type { SearchResultStore } from '@searchspring/snap-store-mobx';
import { MockData } from '@searchspring/snap-shared';
import { SearchMerchandisingStore, SearchPaginationStore, SearchQueryStore } from '@searchspring/snap-store-mobx';
import { UrlManager, UrlTranslator } from '@searchspring/snap-url-manager';

describe('Search Header Component', () => {
	const services = {
		urlManager: new UrlManager(new UrlTranslator()),
	};

	const searchConfig = {
		id: 'search',
	};

	const data = new MockData().searchMeta('dress');

	const paginationStore = new SearchPaginationStore(searchConfig, services, data.pagination, data.meta);
	const queryStore = new SearchQueryStore(services, data.search!);

	it('renders with classname', () => {
		const className = 'classy';
		const rendered = render(<SearchHeader paginationStore={paginationStore} queryStore={queryStore} className={className} />);

		const headerElement = rendered.container.querySelector('.ss__search-header');
		expect(headerElement).toBeInTheDocument();
		expect(headerElement).toHaveClass(className);
	});

	it('can disable styles', () => {
		const rendered = render(<SearchHeader paginationStore={paginationStore} queryStore={queryStore} disableStyles />);

		const headerElement = rendered.container.querySelector('.ss__search-header');

		expect(headerElement?.classList).toHaveLength(1);
	});

	const theme = {
		components: {
			searchHeader: {
				subTitleText: 'hi mom',
			},
		},
	};

	it('is themeable with ThemeProvider', () => {
		const args = {
			paginationStore: paginationStore,
			queryStore: queryStore,
		};
		const rendered = render(
			<ThemeProvider theme={theme}>
				<SearchHeader {...args} />
			</ThemeProvider>
		);

		const subtitle = rendered.container.querySelector('.ss__search-header__subtitle');

		expect(subtitle?.innerHTML).toBe(theme.components.searchHeader.subTitleText);
	});

	it('is themeable with theme prop', () => {
		const args = {
			paginationStore: paginationStore,
			queryStore: queryStore,
		};

		const rendered = render(<SearchHeader {...args} theme={theme} />);
		const subtitle = rendered.container.querySelector('.ss__search-header__subtitle');
		expect(subtitle?.innerHTML).toBe(theme.components.searchHeader.subTitleText);
	});

	it('is themeable with theme prop overrides ThemeProvider', () => {
		const args = {
			paginationStore: paginationStore,
			queryStore: queryStore,
		};

		const componentTheme = {
			components: {
				searchHeader: {
					subTitleText: 'hi dad',
				},
			},
		};

		const rendered = render(
			<ThemeProvider theme={theme}>
				<SearchHeader {...args} theme={componentTheme} />
			</ThemeProvider>
		);

		const subtitle = rendered.container.querySelector('.ss__search-header__subtitle');
		expect(subtitle?.innerHTML).toBe(componentTheme.components.searchHeader.subTitleText);
	});

	it('renders a custom subtitle', async () => {
		const customsubTitle = 'Custom title';
		const rendered = render(<SearchHeader subTitleText={customsubTitle} paginationStore={paginationStore} queryStore={queryStore} />);

		const headerElement = rendered.container.querySelector('.ss__search-header__subtitle');
		expect(headerElement).toBeInTheDocument();
		expect(headerElement).toHaveTextContent(customsubTitle);
	});

	it('renders a custom subtitle using a function', async () => {
		const customsubTitle = (data: any) => {
			const { pagination, search } = data;
			return `${pagination.totalResults} ${search?.query?.string} subtitle!`;
		};
		const rendered = render(<SearchHeader subTitleText={customsubTitle} paginationStore={paginationStore} queryStore={queryStore} />);

		const headerElement = rendered.container.querySelector('.ss__search-header__subtitle');
		expect(headerElement).toBeInTheDocument();
		expect(headerElement).toHaveTextContent('1298 dress subtitle!');
	});

	it('dangerously sets the inner html of subtitle', async () => {
		const customsubTitle = (data: any) => {
			const { pagination, search } = data;
			return `<span class="findMe">${pagination.totalResults} ${search?.query?.string} subtitle!</span`;
		};
		const rendered = render(<SearchHeader subTitleText={customsubTitle} paginationStore={paginationStore} queryStore={queryStore} />);

		const Element = rendered.container.querySelector('.ss__search-header__subtitle .findMe');
		expect(Element).toBeInTheDocument();
		expect(Element).toHaveTextContent('1298 dress subtitle!');
	});

	describe('has Results', () => {
		it('renders the default title', async () => {
			const rendered = render(<SearchHeader paginationStore={paginationStore} queryStore={queryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('Showing 1 - 30 of 1298 results for "dress"');
		});

		it('renders a custom title', async () => {
			const customTitle = 'Custom title';
			const rendered = render(<SearchHeader titleText={customTitle} paginationStore={paginationStore} queryStore={queryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__title');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent(customTitle);
		});

		it('renders a custom title using a function', async () => {
			const customTitle = (data: any) => {
				const { pagination, search } = data;
				return `look mom ${pagination.totalResults} results found for ${search?.query?.string}!`;
			};
			const rendered = render(<SearchHeader titleText={customTitle} paginationStore={paginationStore} queryStore={queryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__title');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('look mom 1298 results found for dress!');
		});

		it('dangerously sets the inner html of title', async () => {
			const customTitle = (data: any) => {
				const { pagination, search } = data;
				return `<span class="findMe">look mom ${pagination.totalResults} results found for ${search?.query?.string}!</span>`;
			};
			const rendered = render(<SearchHeader titleText={customTitle} paginationStore={paginationStore} queryStore={queryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__title .findMe');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('look mom 1298 results found for dress!');
		});
	});

	describe('No Results', () => {
		const noResultsdata = new MockData().searchMeta('noResults');
		const emptyPaginationStore = new SearchPaginationStore(searchConfig, services, noResultsdata.pagination, noResultsdata.meta);
		const emptyQueryStore = new SearchQueryStore(services, noResultsdata.search!);

		it('renders the default no results title', async () => {
			const rendered = render(<SearchHeader paginationStore={emptyPaginationStore} queryStore={emptyQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__noresultstitle');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('No results for "blah" found.');
		});

		it('renders a custom no results title', async () => {
			const customNoResults = 'look mom no results found!';
			const rendered = render(<SearchHeader noResultsText={customNoResults} paginationStore={emptyPaginationStore} queryStore={emptyQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__noresultstitle');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent(customNoResults);
		});

		it('renders a custom no results title using a function', async () => {
			const customNoResults = (data: any) => {
				const { pagination, search } = data;
				return `look mom ${pagination.totalResults} results found for ${search?.query?.string}!`;
			};
			const rendered = render(<SearchHeader noResultsText={customNoResults} paginationStore={emptyPaginationStore} queryStore={emptyQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__noresultstitle');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('look mom 0 results found for blah!');
		});

		it('dangerously sets the inner html of no results title', async () => {
			const customTitle = (data: any) => {
				const { pagination, search } = data;
				return `<span class="findMe">look mom ${pagination.totalResults} results found for ${search?.query?.string}!</span>`;
			};
			const rendered = render(<SearchHeader noResultsText={customTitle} paginationStore={emptyPaginationStore} queryStore={emptyQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__noresultstitle .findMe');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('look mom 0 results found for blah!');
		});
	});

	describe('correctedQueryText', () => {
		const oqData = new MockData().searchMeta('oq');
		const oqPaginationStore = new SearchPaginationStore(searchConfig, services, oqData.pagination, oqData.meta);
		const oqQueryStore = new SearchQueryStore(services, oqData.search!);

		it('renders the default correctedQueryText', async () => {
			const rendered = render(<SearchHeader paginationStore={oqPaginationStore} queryStore={oqQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__corrected');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('No results found for "rodd", showing results for "road" instead.');
		});

		it('renders a custom correctedQueryText', async () => {
			const customoq = 'Custom oq';
			const rendered = render(<SearchHeader correctedQueryText={customoq} paginationStore={oqPaginationStore} queryStore={oqQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__corrected');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent(customoq);
		});

		it('renders a custom correctedQueryText using a function', async () => {
			const customoq = (data: any) => {
				const { pagination, search } = data;
				return `${pagination.totalResults} ${search?.query?.string} oq!`;
			};
			const rendered = render(<SearchHeader correctedQueryText={customoq} paginationStore={oqPaginationStore} queryStore={oqQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__corrected');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('1298 road oq!');
		});

		it('dangerously sets the inner html of correctedQueryText', async () => {
			const customTitle = (data: any) => {
				const { pagination, search } = data;
				return `<span class="findMe">${pagination.totalResults} ${search?.query?.string} oq!</span>`;
			};
			const rendered = render(<SearchHeader correctedQueryText={customTitle} paginationStore={oqPaginationStore} queryStore={oqQueryStore} />);

			const headerElement = rendered.container.querySelector('.ss__search-header__corrected .findMe');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('1298 road oq!');
		});
	});

	describe('landing page', () => {
		const landingData = new MockData().searchMeta('landingPage');
		const landingPaginationStore = new SearchPaginationStore(searchConfig, services, landingData.pagination, landingData.meta);
		const landingQueryStore = new SearchQueryStore(services, landingData.search!);
		const merchandisingStore = new SearchMerchandisingStore(services, landingData.merchandising!);

		console.log(merchandisingStore);
		it('renders the default correctedQueryText', async () => {
			const rendered = render(
				<SearchHeader merchandisingStore={merchandisingStore} paginationStore={landingPaginationStore} queryStore={landingQueryStore} />
			);

			const headerElement = rendered.container.querySelector('.ss__search-header__landingPageTitle');
			expect(headerElement).toBeInTheDocument();
			expect(headerElement).toHaveTextContent('glasses');
		});
	});
});
