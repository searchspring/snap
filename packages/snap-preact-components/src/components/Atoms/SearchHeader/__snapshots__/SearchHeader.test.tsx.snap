// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Header Component No Results dangerously sets the inner html of no results title 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="No results found for blah"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--no-results"
    >
      <span
        class="findMe"
      >
        look mom 0 results found for blah!
      </span>
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component No Results renders a custom no results title 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="No results found for blah"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--no-results"
    >
      look mom no results found!
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component No Results renders a custom no results title using a function 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="No results found for blah"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--no-results"
    >
      look mom 0 results found for blah!
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component No Results renders the default no results title 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="No results found for blah"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--no-results"
    >
      <span>
        
			No results for 
        <span
          class="ss__search-header__results-query"
        >
          "blah"
        </span>
         found.
		
      </span>
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component can disable styles 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component correctedQueryText dangerously sets the inner html of correctedQueryText 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "road"
      </span>
      
	
    </h3>
    <h5
      aria-atomic="true"
      aria-label="No results found for rodd, showing results for road instead"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--corrected"
    >
      <span
        class="findMe"
      >
        1298 road oq!
      </span>
    </h5>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component correctedQueryText renders a custom correctedQueryText 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "road"
      </span>
      
	
    </h3>
    <h5
      aria-atomic="true"
      aria-label="No results found for rodd, showing results for road instead"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--corrected"
    >
      Custom oq
    </h5>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component correctedQueryText renders a custom correctedQueryText using a function 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "road"
      </span>
      
	
    </h3>
    <h5
      aria-atomic="true"
      aria-label="No results found for rodd, showing results for road instead"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--corrected"
    >
      1298 road oq!
    </h5>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component correctedQueryText renders the default correctedQueryText 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "road"
      </span>
      
	
    </h3>
    <h5
      aria-atomic="true"
      aria-label="No results found for rodd, showing results for road instead"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--corrected"
    >
      No results found for 
      <em>
        "rodd"
      </em>
      , showing results for 
      <em>
        "road"
      </em>
       instead.
    </h5>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component dangerously sets the inner html of subtitle 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
    <h4
      aria-atomic="true"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--subtitle"
    >
      <span
        class="findMe"
      >
        1298 dress subtitle!
      </span>
    </h4>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component has Results dangerously sets the inner html of title 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      <span
        class="findMe"
      >
        look mom 1298 results found for dress!
      </span>
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component has Results renders a custom title 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Custom title
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component has Results renders a custom title using a function 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      look mom 1298 results found for dress!
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component has Results renders the default title 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component is themeable with ThemeProvider 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
    <h4
      aria-atomic="true"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--subtitle"
    >
      hi mom
    </h4>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component is themeable with theme prop 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
    <h4
      aria-atomic="true"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--subtitle"
    >
      hi mom
    </h4>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component is themeable with theme prop overrides ThemeProvider 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
    <h4
      aria-atomic="true"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--subtitle"
    >
      hi dad
    </h4>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component landing page renders the default correctedQueryText 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      class="ss__search-header__title ss__search-header__title--landing-page"
    >
      glasses
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component renders a custom subtitle 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
    <h4
      aria-atomic="true"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--subtitle"
    >
      Custom title
    </h4>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component renders a custom subtitle using a function 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
    <h4
      aria-atomic="true"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--subtitle"
    >
      1298 dress subtitle!
    </h4>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component renders with classname 1`] = `
<DocumentFragment>
  <header
    class="ss__search-header classy emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
  </header>
</DocumentFragment>
`;

exports[`Search Header Component uses bocachica theme 1`] = `
<DocumentFragment>
  .emotion-0 .ss__search-header__title {
  color: #202223;
}

<header
    class="ss__search-header emotion-0"
  >
    <h3
      aria-atomic="true"
      aria-label="Now showing 1298 items in the product grid"
      aria-live="polite"
      class="ss__search-header__title ss__search-header__title--results"
    >
      Showing 
      <span
        class="ss__search-header__results-count-range"
      >
         1 - 30 of 
      </span>
       
		
      <span
        class="ss__search-header__results-count-total"
      >
        1298
      </span>
       
		results 
		for 
      <span
        class="ss__search-header__results-query"
      >
        "dress"
      </span>
      
	
    </h3>
  </header>
</DocumentFragment>
`;
